# Statistics 3

***

## Problem Statement 1:
### Blood glucose levels for obese patients have a mean of 100 with a standard deviation of  15. A researcher thinks that a diet high in raw cornstarch will have a positive effect on blood glucose levels. A sample of 36 patients who have tried the raw cornstarch diet have a mean glucose level of 108. Test the hypothesis that the raw cornstarch had an effect or not. 
 
 

# z=(X-μ)/σ

In [1]:
import numpy as np
from scipy import stats
def Zscore(x, mean, stdev):
    return (x - mean)/stdev


x = 108
mean = 100
stdev = 15/np.sqrt(36)

z = Zscore(x, mean, stdev)
print("𝜇= "+str(mean)+"\tX̄= "+str(x)+ "\t𝜎 =  "+str(stdev))
print("z= ",z)
p= stats.norm.cdf(z)
print("p-value associated with 3.20 is ",p)

𝜇= 100	X̄= 108	𝜎 =  2.5
z=  3.2
p-value associated with 3.20 is  0.9993128620620841


#### Performing Hypothesis Testing
Step1:Develop a Hypothesis 
            The population mean is 100.
    
H0: μ = 100
H1: μ ≠ 100
    
Step2:  State significance level:  
        Significance level (α). It is not given in the problem and hence taking it as 5% (0.05).    
    α=0.05
    
Step3: Identify Test Statistic: 
    𝜇=100
    𝜎 = 15
    X̄=108
    n=36
    
    
 z= (108-100) / (15/√36)=3.20
 
Step4: Determine P-value:
         The probability by looking at z- table and p-value associated with 3.20 is 0.9993. 
            Probability of having p-value less than 108 is 0.9993 and more than or equals to 108 is (1-0.9993)=0.0007.


Step5: Compare P-value to Significance level:
        0.0007<0.05.
        Hence  Reject H0,i.e. there is raw cornstarch effect.

***

## Problem Statement 2:

### In one state, 52% of the voters are Republicans, and 48% are Democrats. In a second state, 47% of the voters are Republicans, and 53% are Democrats. Suppose a simple random sample of 100 voters are surveyed from each state. What is the probability that the survey will show a greater percentage of Republican voters in the second state than in the first state?

In [2]:
import numpy as np
from statsmodels.stats.proportion import proportions_ztest
counts = np.array([52,47])
nobs = np.array([100, 100])

zscore, pvalue = proportions_ztest(counts, nobs, alternative = 'larger')
print('zscore = {:.3f}, pvalue = {:.3f}'.format(zscore, pvalue))
print("The probability that the survey will show a greater percentage of Republican voters in the second state than in the first state is 0.24.")

zscore = 0.707, pvalue = 0.240
The probability that the survey will show a greater percentage of Republican voters in the second state than in the first state is 0.24.


####  Alternative method
n1P1 = 100 * 0.52 = 52,

n1(1 - P1) = 100 * 0.48 = 48,

n2P2 = 100 * 0.47 = 47

n2(1 - P2) = 100 * 0.53 = 53 are each greater than 10, the sample size is large enough.

 
Finding the mean of the difference in sample proportions:

E(p1 - p2) = P1 - P2 = 0.52 - 0.47 = 0.05.
 

Find the standard deviation of the difference.

σd =  sqrt{ [ P1(1 - P1) / n1 ] + [ P2(1 - P2) / n2 ] } 

σd = sqrt{ [ (0.52)(0.48) / 100 ] + [ (0.47)(0.53) / 100 ] } 

σd = sqrt (0.002496 + 0.002491) = sqrt(0.004987) = 0.0706

Finding the probability--

This problem requires us to find the probability that p1 is less than p2. 
This is equivalent to finding the probability that p1 - p2 is less than zero. 

To find this probability, we need to transform the random variable (p1 - p2) into a z-score. 

z = (x - μ  ) / σd = = (0 - 0.05)/0.0706 = -0.7082
z-score being -0.7082 or less is 0.24.


Therefore, the probability that the survey will show a greater percentage of Republican voters in the second state than in the first state is 0.24.

## Problem Statement 3:
### You take the SAT and score 1100. The mean score for the SAT is 1026 and the standard deviation is 209. How well did you score on the test compared to the average test taker?

### z=(X-μ)/σ
z=(1100 – 1026) / 209 =0.354. 

A z-score of 0.354 is 0.1368 + 0.5= .6368 or 63.68%.

The z-table shown has scores for the RIGHT of the mean. Therefore, we have to add 0.5 for all of the area LEFT of the mean. 



In [3]:
#Using stats
import numpy as np
from scipy import stats
def Zscore(x, mean, stdev):
    return (x - mean)/stdev


x = 1100
mean = 1026
stdev = 209

z = Zscore(x, mean, stdev)
print("𝜇= "+str(mean)+"\tX̄= "+str(x)+ "\t  𝜎 =  "+str(stdev))
print("z= ",z)
p= stats.norm.cdf(z)
print("p-value associated with "+str(z)+" is ",p)

𝜇= 1026	X̄= 1100	  𝜎 =  209
z=  0.35406698564593303
p-value associated with 0.35406698564593303 is  0.6383556584353189
